``bad-exception-context`` has been renamed to ``bad-exception-cause`` as it is about the cause and not the context.

Closes #3694
